constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -586,16 +586,28 @@ fn combine_to_conditional_assignment<T: FieldElement, V: Ord + Clone + Hash + Eq
         return None;
     }
 
-    Some(ProcessResult {
-        effects: vec![Effect::ConditionalAssignment {
+    let assignment = if let Some(first_assignment_value) = first_assignment.try_to_number() {
+        // We can check the condition now and return an assignment.
+        if rc.allows_value(first_assignment_value) {
+            Effect::Assignment(first_var.clone(), first_assignment.clone())
+        } else {
+            Effect::Assignment(first_var.clone(), second_assignment.clone())
+        }
```

Comment:
I wanted to make a suggestion here that it suffices to use the range constraint of the first assignment instead of `try_to_number`, which is more general.

But then I realized: We should not generate such assignments in the first place. There should be a check in `solve()` above that results in `Err(_)` instead of an assignment, and then we would only have one branch in line 527

Reply:
Oh, interesting. Are you saying I could roll back the changes to this file (except for the test) and everything should still work? And the reason it didn't is perhaps what I fixed in #2708?

Reply:
It looks like it! On `main` the test below fails, on `check-range-constraints` it passes.

Reply:
Haha I only saw 2708 after this comment :)